### ISSUE TITLE
reduce(op, a::StaticArray) is ambiguous

### ISSUE THREAD

**Issue by colinxs** [JuliaArrays/StaticArrays.jl#641]
```
julia> svec = SVector([1,2], [1,2])
julia> reduce(vcat, svec)
ERROR: MethodError: reduce(::typeof(hcat), ::SArray{Tuple{2},Array{Int64,1},1,2}) is ambiguous. Candidates:
  reduce(op, a::StaticArray) in StaticArrays at /home/colinxs/.julia/packages/StaticArrays/VyRz3/src/mapreduce.jl:168
  reduce(::typeof(hcat), A::AbstractArray{#s72,1} where #s72<:(Union{AbstractArray{T,1}, AbstractArray{T,2}} where T)) in Base at abstractarray.jl:1324
Possible fix, define
  reduce(::typeof(hcat), ::StaticArray{S,#s72,1} where S<:Tuple where #s72<:(Union{AbstractArray{T,1}, AbstractArray{T,2}} where T))
```
This is on StaticArrays 0.11 and Julia 1.1.1. As far as I've seen this only happens for vcat/hcat.

I believe it's related to https://github.com/JuliaLang/julia/pull/27188 when specializations were added for vcat and hcat.

**Comment by c42f** [comment]
Yep, this is one of the few remaining ambiguities. I guess we just need to work around it by adding a few specializations.

**Comment by colinxs** [comment]
Thanks for the prompt response! I figured you were already aware but flagged this just in case :).

**Comment by c42f** [comment]
Well I'm not sure we're aware of which ambiguities cause real problems so thanks for reporting it.

**Comment by c42f** [comment]
Fixed in #662, thanks to @mateuszbaran
